core/state_transition.go

```diff
@@ -324,17 +323,24 @@ func (st *StateTransition) preCheck() error {
 			}
 		}
 	}
-
+	// Check that the user is paying at least the current blob fee
 	if st.evm.ChainConfig().IsCancun(st.evm.Context.BlockNumber, st.evm.Context.Time) {
 		if st.blobGasUsed() > 0 {
-			// Check that the user is paying at least the current blob fee
-			blobFee := st.evm.Context.BlobBaseFee
-			if st.msg.BlobGasFeeCap.Cmp(blobFee) < 0 {
-				return fmt.Errorf("%w: address %v have %v want %v", ErrBlobFeeCapTooLow, st.msg.From.Hex(), st.msg.BlobGasFeeCap, blobFee)
+			// Skip the checks if gas fields are zero and blobBaseFee was explicitly disabled (eth_call)
+			if !st.evm.Config.NoBaseFee && msg.BlobGasFeeCap.BitLen() > 0 {
```

Comment:
The way this became changed now, the check is _skipped_ if `msg.BlobGasFeeCap == 0`. And `msg` comes from the transaction we're about to execute. In other words, we omit a check depending on either

1. If a debug-setting is enabled, OR
2. A certain field in a transaction is zero. 

It's not obvious to me why it is safe to do number 2)

Reply:
Shit, good catch. Should have been an ||